tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -14178,6 +14180,35 @@ bool Sema::DiagnoseHLSLDecl(Declarator &D, DeclContext *DC, Expr *BitWidth,
 
   ArBasicKind basicKind = hlslSource->GetTypeElementKind(qt);
 
+  // Check for invalid template params in typed buffer/texture declarations.
```

Comment:
I think this checking needs to be done under `CheckTemplateArgumentListForHLSL`, like by updating `IsValidTemplateArgumentType`, rather than on the `DiagnoseHLSLDecl`.

Reply:
I moved it to `CheckTemplateArgumentListForHLSL`. I was tempted to move it to `IsValidTemplateArgumentType`, but it would need to add some params to that and then would just be an else there instead of outside where we can already identify textures (though I'm not thrilled with how we need to identify them there either)